packages/lib-core/src/store/PluginStore.ts

```diff
@@ -227,11 +228,28 @@ export class PluginStore implements PluginConsumer, PluginManager {
    */
   private isExtensionInUse(extension: Extension) {
     return (
-      (extension.flags?.required?.every((f) => this.options.isFeatureFlagEnabled(f)) ?? true) &&
-      (extension.flags?.disallowed?.every((f) => !this.options.isFeatureFlagEnabled(f)) ?? true)
+      (extension.flags?.required?.every((f) => this.featureFlags[f]) ?? true) &&
+      (extension.flags?.disallowed?.every((f) => !this.featureFlags[f]) ?? true)
     );
   }
 
+  setFeatureFlag = (flagName: string) => (newFlagValue: boolean) => {
+    const prevFeatureFlags = this.featureFlags;
+
+    this.featureFlags = {
+      ...this.featureFlags,
+      [flagName]: newFlagValue,
+    };
+
+    if (!_.isEqual(prevFeatureFlags, this.featureFlags)) {
```

Comment:
@vojtechszocs Since we had discussed supporting 2 states (true/false), shall we check for one of those two states before we update the flag in the store? cc @christianvogt